### ISSUE TITLE
Fed_KiTS19 code generates negative loss values

### ISSUE THREAD

**Issue by akash-07** [owkin/FLamby#276]
Dear authors,

In my attempt to run Fed_KiTS19 code, I observed negative loss value being generated. Is this normal for the loss function used for the Fed_KiTS19 task ? I assume that the `L = (1- Dice) + CE` should not yield negative values. 

I am running the following code:

```python
import torch
from flamby.datasets.fed_kits19 import (
    BATCH_SIZE,
    LR,
    NUM_EPOCHS_POOLED,
    Baseline,
    BaselineLoss,
    NUM_CLIENTS,
    Optimizer,
)
from flamby.datasets.fed_kits19 import FedKits19 as FedDataset

device = torch.device('cuda:0')
train_dataset = FedDataset(center=0, train=True, pooled=False)
n_samples = len(train_dataset)
train_dataloader = torch.utils.data.DataLoader(train_dataset, batch_size=BATCH_SIZE, shuffle=True, num_workers=0)
n_batches =len(train_dataloader)
lossfunc = BaselineLoss()
model = Baseline().to(device)
optimizer = Optimizer(model.parameters(), lr=LR)

# Traditional pytorch training loop
for epoch in range(0, 50):
    epoch_loss = 0
    for idx, (X, y) in enumerate(train_dataloader):
        print(f'Running batch {idx+1}/{n_batches}')
        X = X.to(device)
        y = y.to(device)
        optimizer.zero_grad()
        outputs = model(X)
        loss = lossfunc(outputs, y)
        loss.backward()
        optimizer.step()

        epoch_loss += loss.item() * X.size(0)
    
    avg_epoch_loss = epoch_loss / n_samples
    print(f'Average epoch loss {avg_epoch_loss:.6f}')
```

Last few lines of the output:

```
Average epoch loss -0.098586
Running batch 1/5
Running batch 2/5
Running batch 3/5
Running batch 4/5
Running batch 5/5
Average epoch loss -0.202567
Running batch 1/5
Running batch 2/5
Running batch 3/5
Running batch 4/5
Running batch 5/5
Average epoch loss -0.087862
Running batch 1/5
Running batch 2/5
Running batch 3/5
Running batch 4/5
Running batch 5/5
Average epoch loss -0.160287
```

Any help is highly appreciated.

Thank you !

**Comment by ErumMushtaq** [comment]
The definition of loss function, (DC_and_CE_loss), is slightly different in the nnunet. They have  L = (CE - Dice Score), therefore, it is giving negative values. (https://github.com/MIC-DKFZ/nnUNet/blob/2243aca7ae234c58317a4805c77c12032eb79964/nnunetv2/training/loss/compound_losses.py).

**Comment by jeandut** [comment]
Indeed @akash-07 here is the definition of the dice loss which is returned as -dc: https://github.com/MIC-DKFZ/nnUNet/blob/2243aca7ae234c58317a4805c77c12032eb79964/nnunetv2/training/loss/dice.py#L9

**Comment by jeandut** [comment]
I will add something in the doc and close this issue if that's alright with you @akash-07

**Comment by jeandut** [comment]
Appendix G.5 of the paper is thus incorrect I'll try to update the arxiv.

**Comment by akash-07** [comment]
Thanks @jeandut and @ErumMushtaq for checking out the loss !
